src/locker.rs

```diff
@@ -0,0 +1,112 @@
+use std::ops::{Deref, DerefMut};
+
+use crate::isolate::Isolate;
+use crate::scope::data::ScopeData;
+use crate::support::Opaque;
+
+#[repr(C)]
+#[derive(Debug)]
+struct LockerHandle(Opaque);
+
+/// A handle to a shared isolate, allowing access to the isolate in a thread safe way.
+///
+/// Unlike V8 isolates, these do not currently support re-entrancy.
+/// Do not create multiple lockers to the same isolate in the same thread.
+#[derive(Debug)]
+pub struct Locker<'a> {
+  _lock: raw::Locker,
+  // We maintain a mut reference to ensure we have exclusive ownership of the isolate during the lock.
+  locked: &'a mut Isolate,
+}
+
+impl<'a> Locker<'a> {
+  /// Claims the isolate, this should only be used from a shared isolate.
+  pub(crate) fn new(isolate: &'a mut Isolate) -> Self {
+    let s = Self {
+      _lock: raw::Locker::new(isolate),
+      locked: isolate,
+    };
+    ScopeData::new_root(s.locked);
+    unsafe { s.locked.enter() };
+    s
+  }
+
+  /// Returns a reference to the locked isolate.
+  pub fn isolate(&self) -> &Isolate {
+    self.locked
+  }
+
+  /// Returns a mutable reference to the locked isolate.
+  pub fn isolate_mut(&mut self) -> &mut Isolate {
+    self.locked
+  }
+
+  /// Returns if the isolate is locked by the current thread.
+  pub fn is_locked(isolate: &Isolate) -> bool {
+    raw::Locker::is_locked(isolate)
+  }
+}
+
+impl<'a> Drop for Locker<'a> {
+  fn drop(&mut self) {
+    // A new locker automatically enters the isolate, so be sure to exit the isolate when the locker is exited.
+    unsafe { self.exit() };
+    ScopeData::drop_root(self);
+  }
+}
+
+impl<'a> Deref for Locker<'a> {
+  type Target = Isolate;
+  fn deref(&self) -> &Self::Target {
+    self.isolate()
+  }
+}
+
+impl<'a> DerefMut for Locker<'a> {
+  fn deref_mut(&mut self) -> &mut Self::Target {
+    self.isolate_mut()
+  }
+}
+
+impl<'a> AsMut<Isolate> for Locker<'a> {
+  fn as_mut(&mut self) -> &mut Isolate {
+    self
+  }
+}
+
+mod raw {
+  use std::mem::MaybeUninit;
+
+  use crate::Isolate;
+
+  #[repr(C)]
+  #[derive(Debug)]
+  pub(super) struct Locker([MaybeUninit<usize>; 2]);
```

Comment:
with `src/binding.hpp` and `crate::binding` you can use the actual C++ definition instead of something that happens to hopefully be the correct size.

Reply:
This PR is out of my capabilities, sorry for waste of the time of the review.

I am closing it now.

Reply:
Update: Use binding to set Locker size, but when building rusty_v8 lib, there's an error report cannot find the `v8__Locker__SIZE` I newly added in binding.hpp.